pandas/_libs/parsers.pyx

```diff
@@ -685,10 +685,17 @@ cdef class TextReader:
                     count = counts.get(name, 0)
 
                     if not self.has_mi_columns and self.mangle_dupe_cols:
-                        while count > 0:
-                            counts[name] = count + 1
-                            name = f'{name}.{count}'
-                            count = counts.get(name, 0)
+                        if count > 0:
```

Comment:
would be nice to unify this code between here and the python parser (followon)

Reply:
sounds great!

feels free to open an issue for tracking

Reply:
Though about using #39345 for this